### PR TITLE
Bug 1994222: Add metrics for jaeger-operator

### PR DESCRIPTION
**Pull request by rubenvp8510**
Relevant request and approval can be found at https://docs.google.com/document/d/1DH0OiBgIlBHpO_Thcx0uxb2DFoLYnDSyt4AZVb4KlRw/edit#

Associated BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1994222

Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: Documentation/data-collection.md [hunk 1]
```diff
@@ -560,6 +560,15 @@ data:
     # 99th percentile of the backend commit duration.
     - '{__name__="instance:etcd_disk_backend_commit_duration_seconds:histogram_quantile",quantile="0.99"}'
     #
+    # Number of jaeger instances using certain storage type.
```
Comment (dgrisonnet) [inline Documentation/data-collection.md:563]: Could you add the name of the team that owns these metrics in the same format as the other metrics?

**Comment by dgrisonnet** [comment]
@smarterclayton can you validate the cardinality of these metrics? According to this [document](https://docs.google.com/document/d/1DH0OiBgIlBHpO_Thcx0uxb2DFoLYnDSyt4AZVb4KlRw/edit#), it adds a maximum of 11 series per clusters.

File: Documentation/data-collection.md (hunk 1 above)
Comment (rubenvp8510) [reply Documentation/data-collection.md:563]: What format the team name should have? The team own this metrics will be `@tracing-team`  or is at least the slack name of the team.

**Comment by smarterclayton** [comment]
Approved cardinality of 11

**Comment by rubenvp8510** [comment]
HI, Any insights on why the two first test are failing?

**Comment by dgrisonnet** [comment]
Seems like flakes @rubenvp8510.

/lgtm
/approve

**Comment by dgrisonnet** [comment]
/retitle Bug 1994222: Add metrics for jaeger-operator

**Pull request merged**